exec/java-exec/src/test/java/org/apache/drill/exec/physical/impl/window/TestWindowFrame.java

```diff
@@ -106,6 +106,60 @@ public void testMultipleFramers() throws Exception {
     );
   }
 
+  @Test
+  public void testUnboundedFollowing() throws Exception {
+    testBuilder()
+      .sqlQuery(getFile("window/q3.sql"), TEST_RES_PATH)
+      .ordered()
+      .sqlBaselineQuery(getFile("window/q4.sql"), TEST_RES_PATH)
+      .build()
+      .run();
+  }
+
+  @Test
+  public void testAggregateRowsUnboundedAndCurrentRow() throws Exception {
+    final String table = "dfs_test.`"+TEST_RES_PATH+"/window/b4.p4`";
+    testBuilder()
+      .sqlQuery(getFile("window/aggregate_rows_unbounded_current.sql"), table)
+      .ordered()
+      .sqlBaselineQuery(getFile("window/aggregate_rows_unbounded_current_baseline.sql"), table)
+      .build()
+      .run();
+  }
+
+  @Test
+  public void testLastValueRowsUnboundedAndCurrentRow() throws Exception {
+    final String table = "dfs_test.`"+TEST_RES_PATH+"/window/b4.p4`";
+    testBuilder()
+      .sqlQuery(getFile("window/last_value_rows_unbounded_current.sql"), table)
+      .unOrdered()
+      .sqlBaselineQuery(getFile("window/last_value_rows_unbounded_current_baseline.sql"), table)
+      .build()
+      .run();
+  }
+
+  @Test
+  public void testAggregateRangeCurrentAndCurrent() throws Exception {
```

Comment:
For the RANGE frame specification, are there unit tests that have the same ORDER-BY value repeating across multiple batches ?  This will form the peer group for the range...